### PR TITLE
New branch suggestion

### PR DESCRIPTION
**Pull request by DGEs2018**
@sophialittlejohn could you please re-review all the three files in new-branch suggestion branch ? Meanwhile I will work on my journal as of the 4th chapter. I appreciate it.

### CODE REVIEW

File: Week8/bean-counter.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+		if (string[i] == character) {
+			charQuantity += 1; // with return (charQuantity += 1), it counted repeated characters as single
+		}
+	}
+	return charQuantity;
+}
+function countBs(string) {
+	// this function makes use of the countBs function to get the count of the character.
+	return countChar(string, 'B');
+}
```
Comment (sophialittlejohn) [inline Week8/bean-counter.js]: This function is never being called ⬆️🤔 could you provide an example of how you would call it?

File: Week8/count-bs.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+	for (let i = 0; i < string.length - 1; i++) {
+		if (string[i] === 'B' || string[i] === 'b') {
+			upperCaseBstorer += string[i];
+		}
+	}
+	return upperCaseBstorer.length;
+}
+
+console.log(countBs('Babyboy')); // should return 3
+console.log(countBs('Bamboo')); // should return 2
```
[the remark this answers is not in the document]
Comment (DGEs2018) [reply Week8/count-bs.js]: True, I expect it'd tell me the function by that is already declared. I uncommented each when I had to try the other. But maybe I could rename it.

File: Week8/bean-counter.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
+/*
+Write a function that takes in two string parameters, with the second one containing one of the
+string characters to be counted. 
+*/
+function countChar(string, character) {
+	let charQuantity = 0;
+	for (let i = 0; i < string.length; i++) {
+		if (string[i] == character) {
```
[the remark this answers is not in the document]
Comment (DGEs2018) [reply Week8/bean-counter.js]: OK, I agree, so it compares strictly - both value and type, you mean, right ?

File: Week8/bean-counter.js (hunk 1 above)
Comment (DGEs2018) [reply Week8/bean-counter.js]: Done

File: Week8/is-even.js [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+		return true;
+	}
+	if (num == 1) {
+		return false;
+	} else return isEven(num - 2);
+}
+
+console.log(isEven(40)); // should return true
+console.log(isEven(65)); // should return false
+console.log(isEven(-14)); // should throw 'RangeError: Maximum call stack size exceeded' and terminate the excecution after 10 lines
```
[the remark this answers is not in the document]
Comment (DGEs2018) [reply Week8/is-even.js]: Done

File: Week8/count-bs.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (DGEs2018) [reply Week8/count-bs.js]: Now renamed the second one 'countAllBs'.

**Pull request closed without merging**